src/routes/directions.js

```diff
@@ -1,43 +1,83 @@
 const express = require('express');
-const Directions = require('../models/Directions')
-
+const Directions = require('../models/Directions');
 
 const router = express.Router();
+const directions = new Directions();
+
+// only expose these Google Directions API query parameters for now
+const validQueryParameters = ['origin', 'destination', 'mode', 'alternatives',
+  'arrival_time', 'departure_time', 'transit_mode'];
+
+const validTransportationModes = ['driving', 'walking', 'bicycling', 'transit'];
+
+const validTransitModes = ['bus', 'subway', 'train', 'tram', 'rail'];
 
 router.get('/directions', (req, res) => {
   const request = req.query;
-  let directions = new Directions();
 
-  if (request.origin === undefined || request.origin.length === 0 ||
-    request.destination === undefined || request.destination.length === 0) {
-      return res.sendStatus(400);
+  // remove invalid and empty query parameters
+  Object.keys(request).forEach((parameter) => {
+    if (!validQueryParameters.includes(parameter) || request[parameter] === '') {
+      delete request[parameter];
     }
+  });
 
-    const apiQuery = {
-      origin: request.origin,
-      destination: request.destination,
-      mode: request.mode || 'transit',
-      alternatives: request.alternatives === 'true' || false,
-    };
-
-    if (request.arrival_time === undefined) {
-      if (request.departure_time) {
-        apiQuery.departure_time = request.departure_time;
-      }
-    } else {
-      apiQuery.arrival_time = request.arrival_time;
+  if (request.origin === undefined || request.destination === undefined) {
+    return res
+      .status(400)
+      .set('Content-Type', 'text/plain')
+      .send('The origin and desination parameters are required');
+  }
+
+  if (request.mode) {
+    if (!validTransportationModes.includes(request.mode)) {
+      return res
+        .status(400)
+        .set('Content-Type', 'text/plain')
+        .send('Invalid transportation mode');
     }
+  }
 
-    if (request.mode === 'transit' && request.transit_mode) {
-      apiQuery.transit_mode = request.transit_mode;
+  if (request.transit_mode) {
```

Comment:
@kurtishouser for this kind of situation with an immediately nested `if`, without the need for `else` or `else if`s, I would recommend simply using the `&&` operator as in `if (request.transit_mode && !validTransitModes.includes(request.transit_mode))`